### DISCUSSION TITLE
Document x-disable-provenance usage (Terrafrom/API)

### DISCUSSION THREAD

**Discussion by zumby** [grafana/grafana#63245] [Alerting]
Hi

Please add more detailed documentation on **x-disable-provenance** header usage.

For example, we tried using it in terraform as follows and it does not seem to be working:

```bash
provider "grafana" {
  url  = "https:/example.com"
  auth = "token"
  http_headers = {
    "X-Disable-Provenance" = "true"
  }
}
```

**Comment by olekgo** [comment]
Tried using the header to create UI-editable terraform-provisioned alerts and it did not work for me either. Even when opening the UI with `x-disable-provenance` set.

**Comment by lex-pec** [comment]
Have the same problem, but not in terraform, using Provisioning API in general (I think the point doesn't change).

PUT request `/api/v1/provisioning/alert-rules/{UID}` returns 500 Error:
> { 
"message":"cannot changed provenance from 'api' to '"", 
"traceID":"" 
}

Used `headers = {"X-Disable-Provenance" : "true"}`, instead of `"true"` also tried `"disable"`,`"enable"` etc.

Grafana v9.3.6

**Comment by zumby** [comment]
@lex-pec @olekgo FYI - https://github.com/grafana/grafana/pull/63650
maybe this will make it working

**Reply by lex-pec** [reply]
Yes, thanks. This PR should fix the problem.

**Comment by zumby** [comment]
ok guys this has been resolved. i can confirm that after ANY change to alerts via API/Terraform with the appropriate header for **grafana 9.5.1** the 'Edit' button appears

<img width="328" alt="image" src="https://user-images.githubusercontent.com/26290616/235651802-c0d28b6b-c11a-4536-9f7c-d519ff885293.png">

@lex-pec lex-pec @olekgo

**Reply by lex-pec** [reply]
@zumby, hi.

Upgraded to 9.5.2, but still get an error. May be they fixed that only for TerraForm, but it would be strange since TerraForm uses the same API.
I've tried to make an API call to `/api/v1/provisioning/alert-rules/{UID}` with `{"X-Disable-Provenance" : "disable"}` as a header and still receive that 500 error: 
> "message":"cannot change provenance from 'api' to ''","traceID":""
